### PR TITLE
NRPT-159-2: Add/update publishing logic for mines + QA Feedback

### PR DESCRIPTION
**Pull request by NickPhura**
- Add logic to prevent publishing if fields are missing
  - always allow unpublishing
  - show warning message if mine is unpublished and does not meet publish requirements
  - If the record is already published, deleting a required field will un-check the publish check
- Change save button text and related text strings
- mark link form controls as dirty on drag/drop
  - update test to assert formArray is dirty
- general test updates as needed
- update details publish logic slightly
  - always allow unpublishing
  - only show warning message if mine is unpublished and does not meet publish requirements

Updated with Kits review comments

### CODE REVIEW

File: angular/projects/admin-nrpti/src/app/mines/mines-add-edit/mines-add-edit.component.ts [hunk 1]
```diff
@@ -164,11 +184,56 @@ export class MinesAddEditComponent implements OnInit, OnDestroy {
...
+  checkCanPublish(): boolean {
+    return (
+      this.myForm.get('description').value &&
+      this.myForm.get('summary').value &&
+      this.myForm.get('type').value &&
+      this.mine.name &&
+      this.mine.status &&
+      this.mine.permitNumbers &&
+      this.mine.permitNumbers.length &&
+      this.mine.tailingsImpoundments &&
```
Comment (KitArmstrong) [inline angular/projects/admin-nrpti/src/app/mines/mines-add-edit/mines-add-edit.component.ts:213]: I am not sure that all mines will have tailing impoundments. When they are imported it is possible they get set to '0' which would then fail this check.

File: angular/projects/admin-nrpti/src/assets/styles/components/mine.scss [hunk 2]
```diff
@@ -69,3 +69,10 @@ input[type='text'] {
   font-size: 15px;
   font-style: italic;
 }
+
+.red-subtext {
+  color: $gray6;
```
Comment (KitArmstrong) [inline angular/projects/admin-nrpti/src/assets/styles/components/mine.scss:74]: Color twice here.

File: angular/projects/admin-nrpti/src/app/mines/mines-add-edit/mines-add-edit.component.ts (hunk 1 above)
Comment (NickPhura) [reply angular/projects/admin-nrpti/src/app/mines/mines-add-edit/mines-add-edit.component.ts:213]: Yes, good catch.  Will update.

File: angular/projects/admin-nrpti/src/assets/styles/components/mine.scss (hunk 2 above)
Comment (NickPhura) [reply angular/projects/admin-nrpti/src/assets/styles/components/mine.scss:74]: oops, will fix

File: angular/projects/admin-nrpti/src/app/mines/mines-add-edit/mines-add-edit.component.ts [hunk 3]
```diff
@@ -164,11 +184,58 @@ export class MinesAddEditComponent implements OnInit, OnDestroy {
...
+      this.mine &&
+      this.myForm &&
+      this.myForm.get('description').value &&
+      this.myForm.get('summary').value &&
+      this.myForm.get('type').value &&
+      this.mine.name &&
+      this.mine.status &&
+      this.mine.permitNumbers &&
+      this.mine.permitNumbers.length &&
+      this.mine.tailingsImpoundments &&
```
Comment (KitArmstrong) [inline angular/projects/admin-nrpti/src/app/mines/mines-add-edit/mines-add-edit.component.ts:215]: This one is still going to fail if it is 0.

Comment (NickPhura) [reply angular/projects/admin-nrpti/src/app/mines/mines-add-edit/mines-add-edit.component.ts:215]: Ohh right, sorry I misunderstood what you meant.  Will adjust

Comment (NickPhura) [reply angular/projects/admin-nrpti/src/app/mines/mines-add-edit/mines-add-edit.component.ts]: Updated in latest commit.

I also noticed that our mine.ts was defaulting tailingsImpoundments to null, so I updated that to be 0 instead.

**Pull request merged**